services/cd-service/pkg/repository/repository_test.go

```diff
@@ -677,3 +682,503 @@ func TestRetrySsh(t *testing.T) {
 		})
 	}
 }
+
+type SlowTransformer struct {
+	finished chan struct{}
+	started  chan struct{}
+}
+
+func (s *SlowTransformer) Transform(ctx context.Context, fs billy.Filesystem) (string, error) {
+	s.started <- struct{}{}
+	<-s.finished
+	return "ok", nil
+}
+
+type PanicTransformer struct{}
+
+func (p *PanicTransformer) Transform(ctx context.Context, fs billy.Filesystem) (string, error) {
+	panic("panic tranformer")
+}
+
+var TransformerError = errors.New("error transformer")
+
+type ErrorTransformer struct{}
+
+func (p *ErrorTransformer) Transform(ctx context.Context, fs billy.Filesystem) (string, error) {
+	return "error", TransformerError
+}
+
+type InvalidJsonTransformer struct{}
+
+func (p *InvalidJsonTransformer) Transform(ctx context.Context, fs billy.Filesystem) (string, error) {
+	return "error", invalidJson
+}
+
+func convertToSet(list []uint64) map[int]bool {
+	set := make(map[int]bool)
+	for _, i := range list {
+		set[int(i)] = true
+	}
+	return set
+}
+
+func TestApplyQueuePanic(t *testing.T) {
+	type action struct {
+		CancelBeforeAdd bool
+		CancelAfterAdd  bool
+		Transformer     Transformer
+		// Tests
+		ExpectedError error
+	}
+	tcs := []struct {
+		Name        string
+		Actions     []action
+		ExpectPanic bool
+	}{
+		{
+			Name: "panic at the start",
+			Actions: []action{
+				{
+					Transformer:   &PanicTransformer{},
+					ExpectedError: panicError,
+				}, {
+					ExpectedError: panicError,
+				}, {
+					ExpectedError: panicError,
+				},
+			},
+		},
+		{
+			Name: "panic at the middle",
+			Actions: []action{
+				{
+					ExpectedError: panicError,
+				}, {
+					Transformer:   &PanicTransformer{},
+					ExpectedError: panicError,
+				}, {
+					ExpectedError: panicError,
+				},
+			},
+		},
+		{
+			Name: "panic at the end",
+			Actions: []action{
+				{
+					ExpectedError: panicError,
+				}, {
+					ExpectedError: panicError,
+				}, {
+					Transformer:   &PanicTransformer{},
+					ExpectedError: panicError,
+				},
```

Comment:
these 3 test cases all expect the same error.
Please add a test case that doesn't expect an error, if possible - I guess with no transformer there should also be no error.

Reply:
There are tests cases that do not expect error as part of different test heading, since this one is specifically for panics, didn't add it in.